distribution/test/itests/test-itests-ddf/src/test/java/ddf/test/itests/catalog/TestFederation.java

```diff
@@ -1683,13 +1718,97 @@ private CometDClient setupCometDClient(List<String> channels) throws Exception {
+                        withPostBodyContaining(metacardId))
+                .then(ok(),
+                        contentType("text/xml"),
+                        bytesContent(getCswQueryResponse(metacardId).getBytes()));
+
+        cswServer.whenHttp()
+                .match(Condition.get("/services/csw"),
+                        Condition.parameter("request", "GetRecordById"),
+                        Condition.parameter("id", metacardId))
+                .then(getCswRetrievalHeaders(filename),
+                        chunkedContentWithHeaders(resourceData, Duration.ofMillis(50), 0));
+    }
+
+    /**
+     * Default settings.
+     */
+    private void runParallelInterruptDownloadTest(ClientDownloader client1,
+            ClientDownloader client2) throws Exception {
+        runParallelInterruptDownloadTest(client1, client2, 300, false);
+    }
+
+    /**
+     * Default settings with choice of start order flag.
+     */
+    private void runParallelInterruptDownloadTest(ClientDownloader client1,
+            ClientDownloader client2, boolean flipStartOrder) throws Exception {
+        runParallelInterruptDownloadTest(client1, client2, 300, flipStartOrder);
+    }
+
+    /**
+     * EXECUTION LOGIC FOR PARALLEL INTERRUPTION OF DOWNLOADS
+     * In this template:
+     * We always start client2 first, unless the flipStartOrder flag is set
+     * We always interrupt client1
+     * We always validate the output integrity of client2
+     */
+    private void runParallelInterruptDownloadTest(ClientDownloader client1,
+            ClientDownloader client2, long startDelay, boolean flipStartOrder) throws Exception {
+        String filename = "product1.txt";
+        String metacardId = generateUniqueMetacardId();
+        String resourceData = getResourceData(metacardId);
+
+        setupParallelInterruptDownloadTest(metacardId, filename, resourceData);
+
+        if (flipStartOrder) {
+            client1.startDownload(metacardId);
+            Thread.sleep(startDelay);
+            client2.startDownload(metacardId);
+        } else {
+            client2.startDownload(metacardId);
+            Thread.sleep(startDelay);
+            client1.startDownload(metacardId);
+        }
+
+        Thread.sleep(600);
+
+        client1.stopDownload();
+
+        String result2 = client2.getResults();
+        String result1 = client1.getResults();
+
+        logParallelInterruptTestState("[2287]", resourceData, result1, result2);
+        assertThat("Valid result was not complete", result2, is(resourceData));
+    }
+
+    private void logParallelInterruptTestState(String prefix, String... args) {
+        Iterator<String> arg = Arrays.asList(args)
+                .iterator();
+        System.out.println(String.format("%s Resource Data: %s", prefix, arg.next()));
+        System.out.println(String.format("%s Result 1: %s", prefix, arg.next()));
+        System.out.println(String.format("%s Result 2: %s", prefix, arg.next()));
```

Comment:
I can fix all of this later. I was just not seeing any output from `LOGGER.info(...)` so I defaulted to this so I could make progress.